app/controllers/courses_controller.rb

```diff
@@ -14,8 +14,12 @@ class CoursesController < ApplicationController
 
   def create
     slug_from_params if should_set_slug?
-    @course =
-      Course.create(course_params.merge('passcode' => Course.generate_passcode))
+
```

Comment:
Take this or leave it, but it reads nicer to me to build up `overrides` in parallel.
```ruby
overrides = {}
overrides[:passcode] = Course.generate_passcode
overrides[:type] = ENV['default_course_type'] if ENV['default_course_type']
```